src/shared/containers/Refile/RefileErrorsForm.jsx

```diff
@@ -185,19 +204,28 @@ class RefileErrorsForm extends Component {
   }
 
   /**
-  * @desc Handles sending the form field payload from the state to the proper API endpoint. All
+  * @desc Handle sending the form field payload from the state to the proper API endpoint. All
   * fields are validated prior to executing the ajax call. Updates the form state booleans and
   * result based on successful or error responses
-  * @param {object} event - contains the current event context of the field
+  * @param {boolean} resetOffset - if the request is from a new range of dates, we show the results
+  * from the first page
   */
-  handleFormSubmit(resetDates) {
-    // TODO: execute validations here
-    // Iterate through patron fields and ensure all fields are valid
-    // forIn(this.state.formFields, (value, key) => {
-    //   this.validateField(key, this.state, true);
-    // });
-
-    if (isEmpty(this.state.fieldErrors)) {
+  handleFormSubmit(resetOffset) {
+    // The array that stores the results after validating the date inputs
+    const fieldErrorsArray = [];
+
+    // Iterate through date fields and ensure all fields are valid
+    forIn(this.state.formFields, (value, key) => {
```

Comment:
Could do this with a `forEach`?

(Don't get me wrong, I don't hate lo-dash, just not sure it's so needed anymore — but very happy to discuss, and happy to just leave as it is for now)

Reply:
Maybe we could! I could try testing it.

Reply:
ahhh you know what? `this.state.formFields` is not an array but an object, so we can't use `forEach` with it....

Reply:
Ah, my bad! Thanks for checking.